tests/source.rs

```diff
@@ -314,4 +316,42 @@ mod tests {
         ).unwrap_err();
         assert_err!(err, ErrorKind::JsonPointer);
     }
+
+    #[test]
+    fn json_fmt_test(){
+        let config = Config::new(PathBuf::from("example")).unwrap_or_else(|err| {
+            panic!("Couldn't create the config: {}", err);
+        });
+
+        let result = build(&config, &["json"]);
+
+        if let Err(ref e) = result {
+            panic!("Error: {}", e);
+        }
+
+        let path = Path::new("example/target/doc/data.json");
+
+        let mut file = match File::open(&path) {
+            Err(e) => panic!("couldn't open data.json: {}", e),
+            Ok(file) => file,
+        };
+
+        let mut s = String::new();
+        let result = file.read_to_string(&mut s);
+
+        if let Err(ref e) = result {
+            panic!("Error: {}", e);
+        }
+
+        let result = JsonApiDocument::from_str(&s);
+        let doc = match result{
+            Err(e) => panic!("Error: {}", e),
+            Ok(doc) => doc
+        };
+
+        match doc.validate() {
```

Comment:
How would unwrap_or_else work here? It should return none if no errors are found.